go/test/endtoend/bootstrap_test.go

```diff
@@ -0,0 +1,549 @@
+// Copyright 2025 Supabase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endtoend
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"time"
+
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/multigres/multigres/go/clustermetadata/topo"
+	"github.com/multigres/multigres/go/multiorch/actions"
+	"github.com/multigres/multigres/go/multiorch/coordinator"
+	"github.com/multigres/multigres/go/multipooler/rpcclient"
+	"github.com/multigres/multigres/go/test/utils"
+	"github.com/multigres/multigres/go/tools/pathutil"
+
+	clustermetadatapb "github.com/multigres/multigres/go/pb/clustermetadata"
+	multipoolermanagerpb "github.com/multigres/multigres/go/pb/multipoolermanager"
+	multipoolermanagerdatapb "github.com/multigres/multigres/go/pb/multipoolermanagerdata"
+
+	// Register topo plugins
+	_ "github.com/multigres/multigres/go/plugins/topo"
+)
+
+// nodeInstance represents a multipooler node for bootstrap testing
+type nodeInstance struct {
+	name           string
+	cell           string
+	grpcPort       int
+	pgPort         int
+	pgctldGrpcPort int
+	dataDir        string
+	pgctldProcess  *exec.Cmd
+	multipoolerCmd *exec.Cmd
+}
+
+func TestBootstrapInitialization(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping end-to-end bootstrap test in short mode")
+	}
+
+	if utils.ShouldSkipRealPostgres() {
```

Comment:
`ShouldSkipRealPostgres()` checks of postgres binaries present and `testing.Short()` being true, so if we passed the earlier check for `Short()` above then `ShouldSkipRealPostgres()` will always be false here